src/exporters/s5.py

```diff
@@ -0,0 +1,492 @@
+            ), f"\
+            {pressure_levels} contains invalid pressure levels!\
+            Must be one of:{self.dataset_reference['pressure_level']}"
+
+            if pressure_levels is not None:  # (s0rry) for mypy
+                return {"pressure_level": [str(pl) for pl in pressure_levels]}
+            else:
+                return None
+        else:
+            return None
+
+    def get_valid_variables(self) -> List:
+        """ get `valid_variables` for this S5Exporter object """
+        # print(self.dataset_reference['variable'])
+        return self.dataset_reference["variable"]
+
+    def create_selection_request(
+        self,
+        variable: str,
+        max_leadtime: int,
+        min_year: int = 1993,
+        max_year: int = 2019,
+        min_month: int = 1,
+        max_month: int = 12,
+        selection_request: Optional[Dict] = None,
+    ) -> Dict:
+        """Build up the selection_request dictionary with defaults
+
+        Returns
+        ----------
+        selection_dict: dict
+            A dictionary with all the time-related arguments of the
+            selection dict filled out
+
+        Note:
+        ----
+        - This is a FULL version of the dictionary that is returned. Will likely
+         lead to errors if you don't chunk the data (e.g. by month) before
+         sending the requests to cdsapi
+        - Some attributes are used to create the default arguments
+            (self.product_type, self.granularity, self.dataset, self.pressure_level)
+        """
+        assert (
+            variable in self.dataset_reference["variable"]
+        ), f"Variable: {variable} is not in the valid variables for this \
+            dataset. Valid variables: {self.dataset_reference['variable']}"
+
+        # setup the default selection request
+        processed_selection_request = {
+            "format": "grib",
+            "originating_centre": "ecmwf",
+            "system": "5",
+            "variable": [variable],
+        }
+
+        # add product_type if required
+        if self.product_type is not None:
+            processed_selection_request.update({"product_type": [self.product_type]})
+
+        init_times_dict = self.get_s5_initialisation_times(
+            self.granularity, min_year=min_year, max_year=max_year,
+            min_month=min_month, max_month=max_month
+        )
+
+        for key, val in init_times_dict.items():
+            processed_selection_request[key] = val
+
+        # get the leadtime information
+        leadtimes_dict = self.get_s5_leadtimes(max_leadtime)
+        for key, val in leadtimes_dict.items():
+            processed_selection_request[key] = val
+
+        # by default, we investigate Kenya
+        kenya_region = get_kenya()
+        processed_selection_request["area"] = self.create_area(kenya_region)
+
+        # TODO: do we want this level of flexibility for the user? why add complexity?
```

Comment:
If we are commenting this out, can we remove the input argument from the function? Otherwise, it will be confusing to have a user-defined `selection_request` which we just ignore.

I'm happy for this to be removed from the `ERA5Exporter` too if you think that makes more sense

Reply:
i'm not sure - for this one i think it does (with the current structure like the `all_valid_s5.py` where i know we have captured every use case. the only issue is that some users might want to use our scaffolding (eg the parallelisation) but withtheir preprepared api calls. what do you think?

Reply:
Yea, when I had the selection request argument my main intention was if the user wanted to define their own times, but I think you have handled that by adding the `min/max_year/month` arguments, so its no longer really necessary

Reply:
I can delete it